modules/tensor_mechanics/src/materials/SingleVariableReturnMappingSolution.C

```diff
@@ -220,14 +220,33 @@ SingleVariableReturnMappingSolution::internalSolve(const Real effective_trial_st
         {
           if (scalar_upper_bound != max_permissible_scalar && scalar_lower_bound != 0.0)
           {
-            Real frac = 0.5;
-            scalar_increment =
-                (1.0 - frac) * scalar_lower_bound + frac * scalar_upper_bound - scalar_old;
-            modified_increment = true;
-            if (iter_output)
-              *iter_output << "  Trial scalar_increment exceeded bounds.  Setting between "
-                              "lower/upper bounds. frac: "
-                           << frac << std::endl;
+            if ((residual - residual_old) != 0.0)
```

Comment:
Is this really the right condition? How often is a difference between floating point numbers exactly zero? If this is correct, it probably should include a document stating that the lack of a tolerance check is intended.

Reply:
Just trying to avoid a divide by zero.  The same check is elsewhere in this file.  I'll add a tolerance.

Reply:
Yeah, just pointing out that exact comparisons with floating point numbers rarely occur with real numbers. Those other places are equally prone to logic issues and I'd like to remove them. Thanks for addressing this comment.